doc/mpc_node_specification.md

```diff
@@ -0,0 +1,225 @@
+    - Transition to `Prepare` state
+- Once all INIT messages have been answered with a success response
+    - Wait for finality of the request (for signatures only)
+    - Send out a START message all participants
+    - Transition to `Running` state
+    - Start poking own generator and sending out the corresponding messages
+
+Owners must also handle the following error case:
+- Not all participants answered the INIT message with an HTTP success response
+    - Select a new set of participants and send another round of INIT messages
+        - Participants must overwrite the previously stored participants list
+    - Wait again for all participants to answer
+
+Other error cases can be handled with a global timeout to abort and retry for
+signatures.
+
+## Triple and Pre-Signature State Machine
+
+Triples and Pre-Signatures are persistently stored on each node. They go through
+the following lifecycle. Nodes must keep track of them in order for state sync
+to work.
+
+![graph](./graphs/p_and_t_states.svg)
+
+### States
+
+`init` and `done` are implicit states. Both are represented by not having any
+information about the P or T in storage. All other states must be explicitly
+stored and ideally persisted.
+
+#### Generating
+
+A P or T in the `Generating` state has an active protocol invocation that will
+generate it on success. It is treated the same way as `Available` for state
+sync. But the storage does not actually hold the P or T, yet.
+
+#### Available
+
+In the `Available` phase, a P or T share is currently ready to be used in a
+protocol. This state always implies that the P or T is available in state. As
+soon as it is removed from persistent storage, we must persist the state change
+to `Using`.
+
+#### Using
+
+A P or T in the `Using` state is currently being used in a protocol invocation.
+We need to track this in order for state sync to not delete any data that's
+still in use.
+
+Only transition from `Using` to `done` when either the protocol finishes
+successfully, or we hit the global protocol timeout. Once that transition
+happens, the node can delete all info about the P or T and its ID from
+persistent storage.
+
+
+## Participant Selection
+
+When a proposer selects a list of participants for a new protocol invocation, it should check these conditions:
+
+- Participants must be online
+- Participants must hold the relevant P and T shares
+
+To enable the first, nodes keep a connection status for every peer. For the
+second part, they keep a directory of each owned P and T with a list of share
+holders. Both data structures are further specified below.
+
+### Node Connection Status
+
+![graph](./graphs/node_connection.svg)
+
+Nodes keep pinging each other, checking their peers status.
+
+Peers that don't respond are marked as `Offline`.
+
+Peers that respond but indicate in their status response that they are not actively participating in the MPC network, are marked as `Inactive`.
```

Comment:
What are the reasons to be Inactive? Can we skip it and use Syncing only?

Reply:
We already have the inactive state in the code. As I understand it, these are for nodes that are joining the network. I don't think we have anything to sync with them, yet.